### DISCUSSION TITLE
Gatsby Source Plugin for Directus 9.x

### DISCUSSION THREAD

**Discussion by sapkra** [directus/directus#3643] [Feature Requests]
There has been a plugin for 8.x but it seems to be deprecated after the rewrite.

Discussed it in https://github.com/directus/directus/issues/2529 already and created this discussion to have it on track.

**Comment by jooola** [comment]
The new plugins for v9 doesn't necessarily need a lot work, since v9 has graphql embedded (correct me if I'm wrong). So the plugin for v8 should work with directus v9 (did the API changed between v8 and v9 ?) but a rewrite should happen on the plugin side to leverage the power of graphql.

Not sure what the future of this plugin is, we need to get input from @loganwishartcraig to see if he has time to restore this plugin. I don't think I will have that much time. 
Directus need to recreate a repo for this plugin, otherwise the plugin will not benefit from having a joint effort toward a single gatsby plugin under the Directus umbrella.

**Reply by rijkvanzanten** [reply]
> Directus need to recreate a repo for this plugin

We'll manage it under `/packages` in the monorepo directus/directus 👍🏻

**Reply by loganwishartcraig** [reply]
Hey! 

I haven't really tested the plugin with v9 so I can't comment on functionality. 

At work we're in the process of migrating to v9 and have been testing out [gatsby-source-graphql](https://www.gatsbyjs.com/plugins/gatsby-source-graphql/) directed to the Directus gql endpoint (shout out to @rijkvanzanten  and team on v9 - it's awesome!). So far, it has suited our needs well enough.

As for future development of a dedicated Gatsby plugin - unfortunately I won't have the time to migrate it to v9 unless my work signs off on it, and that would only happen if we run into a critical roadblock with the current gatsby-source-graphql approach. If something changes here, I'll be sure to notify.

As @jooola said, there's a lot of opportunity for the plugin to piggy-back off of the exposed graphql endpoint instead of the current strategy.

**Reply by benhaynes** [reply]
Thanks @loganwishartcraig! Let us know how things go... in the meantime, @WoLfulus is working on a new integration! 🎉

**Reply by WoLfulus** [reply]
leveraging stuff from `gatsby-source-graphql` is what I'm going for actually! we're still building our own to make some stuff easier for authentication, but should work really close to what that package provides now

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
